src/registry/radio/helpers.ts

```diff
@@ -9,3 +10,27 @@ export const checkIsManualOptions = (
 ): component is RadioComponentSchema & {values: Option[] | undefined} => {
   return component.openForms.dataSrc === 'manual';
 };
+
+// A type guard is needed because TS cannot figure out it's a discriminated union
+// when the discriminator is nested.
+// See https://github.com/microsoft/TypeScript/issues/18758
+export const checkIsReferentielijstenOptions = (
+  component: RadioComponentSchema
+): component is RadioComponentSchema & {
+  data: {values: Option[] | undefined};
+  openForms: {code: string; service: string};
+} => {
+  return component.openForms.dataSrc === 'referentielijsten';
+};
+
+// A type guard is needed because TS cannot figure out it's a discriminated union
+// when the discriminator is nested.
+// See https://github.com/microsoft/TypeScript/issues/18758
+export const checkIsVariableOptions = (
+  component: RadioComponentSchema
+): component is RadioComponentSchema & {
+  data: {values: Option[] | undefined};
```

Comment:
```suggestion
```
same as above - the `values` will not be set since those are sorted out by the backend dynamically (which effectively transforms the component into dataSrc "manual")